File: test-fixtures/src/test/java/polyglot/polyglot-helper.js
/**
 * JavaScript helper for polyglot debugging test.
 * Set breakpoints here to test JavaScript debugging.
 */
(function() {

    function calculateTotal(items) {
        // BREAKPOINT HERE - test JS debugging
        var total = 0;

        for (var i = 0; i < items.length; i++) {
            var item = items[i];
            var subtotal = item.price * item.quantity;
            total = total + subtotal;
        }

        return total;
    }

    function formatCurrency(amount) {
        // BREAKPOINT HERE - test JS debugging
        return '$' + amount.toFixed(2);
    }

    function processOrder(order) {
        // BREAKPOINT HERE - test JS debugging
        var items = order.items;
        var total = calculateTotal(items);
        var formatted = formatCurrency(total);

        return {
            orderId: order.id,
            itemCount: items.length,
            total: total,
            formattedTotal: formatted,
            status: 'processed'
        };
    }

    return {
        calculateTotal: calculateTotal,
        formatCurrency: formatCurrency,
        processOrder: processOrder
    };
})()

File: test-fixtures/src/test/java/polyglot/PolyglotHelper.java
package polyglot;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * Java helper class for polyglot debugging test.
 * Set breakpoints here to test Java debugging.
 */
public class PolyglotHelper {

    /**
     * Validate an order and return validation result.
     */
    public static Map<String, Object> validateOrder(Map<String, Object> order) {
        // BREAKPOINT HERE - test Java debugging (line 18)
        Map<String, Object> result = new HashMap<>();
        
        // Check if order has required fields
        boolean hasId = order.containsKey("id");
        boolean hasItems = order.containsKey("items");
        
        result.put("valid", hasId && hasItems);
        result.put("hasId", hasId);
        result.put("hasItems", hasItems);
        
        if (hasItems) {
            List<?> items = (List<?>) order.get("items");
            result.put("itemCount", items.size());
        } else {
            result.put("itemCount", 0);
        }
        
        return result;
    }

    /**
     * Calculate tax for an amount.
     * BREAKPOINT HERE - test Java debugging
     */
    public static double calculateTax(double amount, double taxRate) {
        double tax = amount * taxRate;
        double total = amount + tax;
        return Math.round(total * 100.0) / 100.0;
    }

    /**
     * Generate an order confirmation.
     * BREAKPOINT HERE - test Java debugging
     */
    public static Map<String, Object> generateConfirmation(String orderId, double total) {
        Map<String, Object> confirmation = new HashMap<>();
        
        confirmation.put("orderId", orderId);
        confirmation.put("total", total);
        confirmation.put("confirmationNumber", "CONF-" + System.currentTimeMillis());
        confirmation.put("status", "confirmed");
        
        return confirmation;
    }
}

